xarray/core/duck_array_ops.py

```diff
@@ -72,16 +50,15 @@ def fail_on_dask_array_input(values, msg=None, func_name=None):
         raise NotImplementedError(msg % func_name)
 
 
-around = _dask_or_eager_func("around")
-isclose = _dask_or_eager_func("isclose")
-
-
-isnat = np.isnat
-isnan = _dask_or_eager_func("isnan")
-zeros_like = _dask_or_eager_func("zeros_like")
+# Requires special-casing because pandas won't automatically dispatch to dask.isnull via NEP-18
+def _dask_or_eager_isnull(obj):
+    if is_duck_dask_array(obj):
```

Comment:
does `dask_array.isnull` dispatch for duck dask arrays, or does this also create e.g. `dask(pint(dask))` objects?

Reply:
It does this:

```python
d = dask_array.array([1, 2, 3])
q = pint.Quantity(d, units='m')
dask.array.isnull(q)
```
raises warnings
```
/home/tegn500/miniconda3/envs/xarray-testing-min-all-deps-py37/lib/python3.7/site-packages/dask/array/core.py:2756: UserWarning: Passing an object to dask.array.from_array which is already a Dask collection. This can lead to unexpected behavior.
  "Passing an object to dask.array.from_array which is already a "
/home/tegn500/miniconda3/envs/xarray-testing-min-all-deps-py37/lib/python3.7/site-packages/numpy/core/_asarray.py:85: UnitStrippedWarning: The unit of the quantity is stripped when downcasting to ndarray.
  return array(a, dtype, copy=False, order=order)
```
and returns the result
```
dask.array<_asarray_isnull, shape=(3,), dtype=bool, chunksize=(3,), chunktype=numpy.ndarray>
```

I think that's a sensible result of calling `isnull` on a chunked pint array?